packages/batch-tx-handler/src/l1-batch-handler/utils.ts

```diff
@@ -184,19 +183,31 @@ export const getRawData = async (sequencerTx: string): Promise<[Uint8Array, BigN
 };
 
 const processDASBatch = async (rawData: Uint8Array) => {
-  const req = await fetch(NovaDacUrl);
+  if(!process.env.NovaDacListUrl) {
+    throw new Error("You are calling anytrust dac while don't provide the dac list url")
+  }
+  const req = await fetch(process.env.NovaDacListUrl);
   const urls = (await req.text()).split('\n');
   if (urls.length === 0) {
     throw Error('No online das servers now');
   }
-  return getDACData(urls[0], rawData);
+  return getDACData(urls, rawData);
 };
 
-const getDACData = async (url: string, rawData: Uint8Array) => {
+// Here is the reference in nitro source code: https://github.com/OffchainLabs/nitro/blob/v2.1.3/arbstate/inbox.go#L127
+const getDACData = async (urls: string[], rawData: Uint8Array) => {
   // The first byte is header flag, the 2nd to 33rd bytes is keyset hash, 34th to 65th is data hash which is what we want.
   const dataHash = ethers.utils.hexlify(rawData.subarray(33, 65));
-  const requestUrl = url + `/get-by-hash/` + dataHash.substring(2);
-  const req = await fetch(requestUrl);
+  let req;
+  for(let i = 0; i < urls.length; i++) {
+    const requestUrl = urls[i] + `/get-by-hash/` + dataHash.substring(2);
+    try {
+      req = await fetch(requestUrl);
+    } catch {
+      console.log(`URL for one of the da node (${urls[i]}) is broken, trying another one...`);
+    }
+    break;
+  }
```

Comment:
i think you'd want to put the `break;` in the `try` block after `req = ...` so that the loop breaks when the fetch request is successful, and continues when the fetch failed?

additionally, the way this code is written, it's assuming that at least one url must return the fetch request result successfully

with the current approach, if none of requestUrls work, and none of the `req` return the correct info, `base64Data` and `base64Data.data` will be `undefined` and so this function would return the wrong result

Reply:
other than that the code on this PR looks good

so feel free to merge and handle this on a later PR